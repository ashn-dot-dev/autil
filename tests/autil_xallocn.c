#define AUTIL_IMPLEMENTATION
#include "../autil.h"

int
main(void)
{
    size_t const nmemb = 4096;
    int* const mem = autil_xallocn(NULL, nmemb, sizeof(int));
    for (size_t i = 0; i < nmemb; ++i) {
        // Set each element to some value just to make source touching that
        // address doesn't cause a segfault.
        mem[i] = (int)i;
    }
    // Free memory.
    autil_xallocn(mem, 0, AUTIL_XALLOC_FREE);

    // Crash the program by asking a total amount of memory that would overflow
    // when nmemb and size are multiplied together.
    autil_xallocn(NULL, (size_t)-1, sizeof(long));

    puts("UNREACHABLE");
    return EXIT_SUCCESS;
}
