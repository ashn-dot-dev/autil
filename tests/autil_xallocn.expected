error: [autil_xallocn] Integer overflow
